### PR TITLE
document rotation of cloudprovider secret

### PR DESCRIPTION
**Pull request by BeckerMax**
**How to categorize this PR?**

/area documentation
 
**What this PR does / why we need it**:
See https://github.com/gardener/gardener/issues/4630

I thought we could start a page where we list relevant secrets and how to rotate them. Especially, as we plan to invest more in the topic of secret rotation we are going to need a place for documenting that.

**Note for reviewer**:

The doc reflects my current understanding which might be incomplete or partially incorrect, if this is the case even regarding minor corrections please comment. Thanks!

```other operator
[add documentation](https://github.com/gardener/gardener/blob/master/docs/usage/secrets_rotation.md#user-provided-secrets) on rotation of cloud provider secret
```

### CODE REVIEW

File: docs/usage/secrets_rotation.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+*Example*: https://github.com/gardener/gardener/blob/master/example/70-secret-provider.yaml
+
+*Usage*: Authenticate gardener and kubernetes components for infrastructure operations
+
+*Rotation*: Rotating the cloudprovider secret requires multiple steps:
+
+1. Either create the new secret in the garden cluster and add a secretBinding or edit the current secret.
+2. :warning: Wait until all Shoots using the secret are reconciled before you deactivate the old secret! Otherwise the shoots will no longer function.
+3. After you are sure all Shoots using the secret are reconciled you can go ahead and deactivate the old secret in your infrastructure provider.
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]: Can you make it more clear what you mean with "deactivate the old secret"?

File: docs/usage/secrets_rotation.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+*Description*: Gardener uses the cloudprovider secret to interact with the infrastructure when setting up shoot networks or security groups via the [terraformer](https://github.com/gardener/terraformer). It is also used by the cloud-controller-manager of your Shoot for example to create VMs. Depending on the cloudprovider the format of the secret differs. Please consult the example above and respective infrastructure extension documentation for the concrete layout.
```
Comment (rfranzke) [inline docs/usage/secrets_rotation.md]: Move this before the rotation topic?

Comment (timuthy) [inline docs/usage/secrets_rotation.md]: Can you amend the example about CCM? I'm not aware that it creates VMs (that's MCM's job).
https://kubernetes.io/docs/concepts/architecture/cloud-controller/

Comment (BeckerMax) [reply docs/usage/secrets_rotation.md]: Yes, thanks for the remark. I think I got it now: The MCM creates the VMs. The CCM then is only interacting with Node objects (Node controller). I adapted the description. PTAL

File: docs/usage/secrets_rotation.md (hunk 1 above)
Comment (BeckerMax) [reply docs/usage/secrets_rotation.md]: adapted it PTAL

File: docs/usage/secrets_rotation.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+**Cloudprovider Secret**
+
+*Example*: https://github.com/gardener/gardener/blob/master/example/70-secret-provider.yaml
+
+*Usage*: Authenticate gardener and kubernetes components for infrastructure operations
+
+*Description*: Gardener uses the cloudprovider secret to interact with the infrastructure when setting up shoot networks or security groups via the [terraformer](https://github.com/gardener/terraformer). It is also used by the [cloud-controller-manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/) of your Shoot to communicate with the infrastructure for example to create Loadbalancer services, routes or retrieve information about Node objects.
+Depending on the cloudprovider the format of the secret differs. Please consult the example above and respective infrastructure extension documentation for the concrete layout.
+
+To put it in use a cloudprovider secret is bound to one more namespaces (and therefore projects) using a [SecretBinding](https://github.com/gardener/gardener/blob/master/example/80-secretbinding.yaml). For Shoots created in those projects the secret is synced to the shoot namespace in the seed cluster.
```
Comment (timuthy) [inline docs/usage/secrets_rotation.md]:
```suggestion
To put it in use a cloudprovider secret is bound to one or more namespaces (and therefore projects) using a [SecretBinding](https://github.com/gardener/gardener/blob/master/example/80-secretbinding.yaml). For Shoots created in those projects the secret is synced to the shoot namespace in the seed cluster.
```

I assume you mean that?

**Review by timuthy — CHANGES_REQUESTED** [review]

**Review by rfranzke — APPROVED** [review]

**Review by timuthy — APPROVED** [review]
/lgtm

**Comment by timebertt** [comment]
@BeckerMax please add a link to the document to the release note, so that users can easily find it.

**Pull request merged**